{"created": 1649925246.4473252, "duration": 0.2825312614440918, "exitcode": 1, "root": "C:\\Users\\Coke\\playground\\Section3\\ds-sa-db-api", "environment": {"Python": "3.8.13", "Platform": "Windows-10-10.0.19043-SP0", "Packages": {"pytest": "6.2.2", "py": "1.10.0", "pluggy": "0.13.1"}, "Plugins": {"json-report": "1.5.0", "metadata": "2.0.1", "submit": "1.3.3"}}, "summary": {"failed": 12, "error": 2, "total": 14, "collected": 14}, "collectors": [{"nodeid": "", "outcome": "passed", "result": [{"nodeid": "tests/__init__.py", "type": "Package"}, {"nodeid": "tests/Part_1/test_Part_1.py", "type": "Module"}, {"nodeid": "tests/Part_2/test_Part_2.py", "type": "Module"}, {"nodeid": "tests/Part_3/test_Part_3.py", "type": "Module"}, {"nodeid": "tests/Part_4/__init__.py", "type": "Package"}]}, {"nodeid": "tests/__init__.py", "outcome": "passed", "result": []}, {"nodeid": "tests/conftest.py", "outcome": "passed", "result": []}, {"nodeid": "tests/__init__.py", "outcome": "passed", "result": [{"nodeid": "tests/__init__.py", "type": "Module"}, {"nodeid": "tests/conftest.py", "type": "Module"}]}, {"nodeid": "tests/Part_1/test_Part_1.py", "outcome": "passed", "result": [{"nodeid": "tests/Part_1/test_Part_1.py::test_Part_1_exist", "type": "Function", "lineno": 13}, {"nodeid": "tests/Part_1/test_Part_1.py::test_Part_1_schema", "type": "Function", "lineno": 21}, {"nodeid": "tests/Part_1/test_Part_1.py::test_Part_1_count", "type": "Function", "lineno": 36}, {"nodeid": "tests/Part_1/test_Part_1.py::test_Part_1_data", "type": "Function", "lineno": 45}]}, {"nodeid": "tests/Part_2/test_Part_2.py", "outcome": "passed", "result": [{"nodeid": "tests/Part_2/test_Part_2.py::test_Part_2_exist", "type": "Function", "lineno": 13}, {"nodeid": "tests/Part_2/test_Part_2.py::test_Part_2_schema", "type": "Function", "lineno": 21}, {"nodeid": "tests/Part_2/test_Part_2.py::test_Part_2_count", "type": "Function", "lineno": 36}, {"nodeid": "tests/Part_2/test_Part_2.py::test_Part_2_data", "type": "Function", "lineno": 45}]}, {"nodeid": "tests/Part_3/test_Part_3.py", "outcome": "passed", "result": [{"nodeid": "tests/Part_3/test_Part_3.py::test_Part_3_exist", "type": "Function", "lineno": 13}, {"nodeid": "tests/Part_3/test_Part_3.py::test_Part_3_schema", "type": "Function", "lineno": 21}, {"nodeid": "tests/Part_3/test_Part_3.py::test_Part_3_count", "type": "Function", "lineno": 36}, {"nodeid": "tests/Part_3/test_Part_3.py::test_Part_3_data", "type": "Function", "lineno": 45}]}, {"nodeid": "tests/Part_4/__init__.py", "outcome": "passed", "result": []}, {"nodeid": "tests/Part_4/conftest.py", "outcome": "passed", "result": []}, {"nodeid": "tests/Part_4/test_Part_4.py", "outcome": "passed", "result": [{"nodeid": "tests/Part_4/test_Part_4.py::test_passenger_table", "type": "Function", "lineno": 4}, {"nodeid": "tests/Part_4/test_Part_4.py::test_titanic_data", "type": "Function", "lineno": 30}]}, {"nodeid": "tests/Part_4/__init__.py", "outcome": "passed", "result": [{"nodeid": "tests/Part_4/__init__.py", "type": "Module"}, {"nodeid": "tests/Part_4/conftest.py", "type": "Module"}, {"nodeid": "tests/Part_4/test_Part_4.py", "type": "Module"}]}], "tests": [{"nodeid": "tests/Part_1/test_Part_1.py::test_Part_1_exist", "lineno": 13, "outcome": "failed", "keywords": ["ds-sa-db-api", "tests/Part_1/test_Part_1.py", "test_Part_1_exist"], "setup": {"duration": 0.0008103000000000415, "outcome": "passed", "stdout": "\n        -------------------------------------\n        Setting up...\n        -------------------------------------\n        \n"}, "call": {"duration": 0.00036800000000003497, "outcome": "failed", "crash": {"path": "C:\\Users\\Coke\\playground\\Section3\\ds-sa-db-api\\tests\\Part_1\\test_Part_1.py", "lineno": 15, "message": "assert False"}, "traceback": [{"path": "tests\\Part_1\\test_Part_1.py", "lineno": 15, "message": "AssertionError"}], "longrepr": "def test_Part_1_exist():\n>       assert table_exist\nE       assert False\n\ntests\\Part_1\\test_Part_1.py:15: AssertionError"}, "teardown": {"duration": 0.0003018000000000187, "outcome": "passed"}}, {"nodeid": "tests/Part_1/test_Part_1.py::test_Part_1_schema", "lineno": 21, "outcome": "failed", "keywords": ["ds-sa-db-api", "test_Part_1_schema", "tests/Part_1/test_Part_1.py"], "setup": {"duration": 0.0002474999999999561, "outcome": "passed"}, "call": {"duration": 0.00022540000000004223, "outcome": "failed", "crash": {"path": "C:\\Users\\Coke\\playground\\Section3\\ds-sa-db-api\\tests\\Part_1\\test_Part_1.py", "lineno": 29, "message": "AttributeError: 'NoneType' object has no attribute 'cursor'"}, "traceback": [{"path": "tests\\Part_1\\test_Part_1.py", "lineno": 29, "message": "AttributeError"}], "longrepr": "def test_Part_1_schema():\n        fields = {\n            ('AlbumId','INTEGER', 1, 1),\n            ('Title','NVARCHAR(160)', 0, 0),\n            ('ArtistId','INTEGER', 0, 0)\n        }\n    \n>       cursor = conn.cursor()\nE       AttributeError: 'NoneType' object has no attribute 'cursor'\n\ntests\\Part_1\\test_Part_1.py:29: AttributeError"}, "teardown": {"duration": 0.00031639999999999446, "outcome": "passed"}}, {"nodeid": "tests/Part_1/test_Part_1.py::test_Part_1_count", "lineno": 36, "outcome": "failed", "keywords": ["ds-sa-db-api", "test_Part_1_count", "tests/Part_1/test_Part_1.py"], "setup": {"duration": 0.00031519999999995996, "outcome": "passed"}, "call": {"duration": 0.00023099999999998122, "outcome": "failed", "crash": {"path": "C:\\Users\\Coke\\playground\\Section3\\ds-sa-db-api\\tests\\Part_1\\test_Part_1.py", "lineno": 39, "message": "AttributeError: 'NoneType' object has no attribute 'cursor'"}, "traceback": [{"path": "tests\\Part_1\\test_Part_1.py", "lineno": 39, "message": "AttributeError"}], "longrepr": "def test_Part_1_count():\n    \n>       cursor = conn.cursor()\nE       AttributeError: 'NoneType' object has no attribute 'cursor'\n\ntests\\Part_1\\test_Part_1.py:39: AttributeError"}, "teardown": {"duration": 0.00029489999999998684, "outcome": "passed"}}, {"nodeid": "tests/Part_1/test_Part_1.py::test_Part_1_data", "lineno": 45, "outcome": "failed", "keywords": ["ds-sa-db-api", "tests/Part_1/test_Part_1.py", "test_Part_1_data"], "setup": {"duration": 0.0003159000000000356, "outcome": "passed"}, "call": {"duration": 0.0002653999999999712, "outcome": "failed", "crash": {"path": "C:\\Users\\Coke\\playground\\Section3\\ds-sa-db-api\\tests\\Part_1\\test_Part_1.py", "lineno": 48, "message": "AttributeError: 'NoneType' object has no attribute 'cursor'"}, "traceback": [{"path": "tests\\Part_1\\test_Part_1.py", "lineno": 48, "message": "AttributeError"}], "longrepr": "def test_Part_1_data():\n    \n>       cursor = conn.cursor()\nE       AttributeError: 'NoneType' object has no attribute 'cursor'\n\ntests\\Part_1\\test_Part_1.py:48: AttributeError"}, "teardown": {"duration": 0.0005353999999999637, "outcome": "passed", "stdout": "\n        -------------------------------------\n        Tearing down...\n        -------------------------------------\n        \n"}}, {"nodeid": "tests/Part_2/test_Part_2.py::test_Part_2_exist", "lineno": 13, "outcome": "failed", "keywords": ["ds-sa-db-api", "test_Part_2_exist", "tests/Part_2/test_Part_2.py"], "setup": {"duration": 0.00041650000000004184, "outcome": "passed", "stdout": "\n        -------------------------------------\n        Setting up...\n        -------------------------------------\n        \n"}, "call": {"duration": 0.0003084000000000975, "outcome": "failed", "crash": {"path": "C:\\Users\\Coke\\playground\\Section3\\ds-sa-db-api\\tests\\Part_2\\test_Part_2.py", "lineno": 15, "message": "assert False"}, "traceback": [{"path": "tests\\Part_2\\test_Part_2.py", "lineno": 15, "message": "AssertionError"}], "longrepr": "def test_Part_2_exist():\n>       assert table_exist\nE       assert False\n\ntests\\Part_2\\test_Part_2.py:15: AssertionError"}, "teardown": {"duration": 0.00024049999999997684, "outcome": "passed"}}, {"nodeid": "tests/Part_2/test_Part_2.py::test_Part_2_schema", "lineno": 21, "outcome": "failed", "keywords": ["test_Part_2_schema", "ds-sa-db-api", "tests/Part_2/test_Part_2.py"], "setup": {"duration": 0.0002898000000000067, "outcome": "passed"}, "call": {"duration": 0.00023159999999999847, "outcome": "failed", "crash": {"path": "C:\\Users\\Coke\\playground\\Section3\\ds-sa-db-api\\tests\\Part_2\\test_Part_2.py", "lineno": 29, "message": "AttributeError: 'NoneType' object has no attribute 'cursor'"}, "traceback": [{"path": "tests\\Part_2\\test_Part_2.py", "lineno": 29, "message": "AttributeError"}], "longrepr": "def test_Part_2_schema():\n        fields = {\n            ('AlbumId','INTEGER', 1, 1),\n            ('Title','NVARCHAR(160)', 0, 0),\n            ('ArtistId','INTEGER', 0, 0)\n        }\n    \n>       cursor = conn.cursor()\nE       AttributeError: 'NoneType' object has no attribute 'cursor'\n\ntests\\Part_2\\test_Part_2.py:29: AttributeError"}, "teardown": {"duration": 0.000302400000000036, "outcome": "passed"}}, {"nodeid": "tests/Part_2/test_Part_2.py::test_Part_2_count", "lineno": 36, "outcome": "failed", "keywords": ["ds-sa-db-api", "tests/Part_2/test_Part_2.py", "test_Part_2_count"], "setup": {"duration": 0.0004199000000000286, "outcome": "passed"}, "call": {"duration": 0.0003130999999999551, "outcome": "failed", "crash": {"path": "C:\\Users\\Coke\\playground\\Section3\\ds-sa-db-api\\tests\\Part_2\\test_Part_2.py", "lineno": 39, "message": "AttributeError: 'NoneType' object has no attribute 'cursor'"}, "traceback": [{"path": "tests\\Part_2\\test_Part_2.py", "lineno": 39, "message": "AttributeError"}], "longrepr": "def test_Part_2_count():\n    \n>       cursor = conn.cursor()\nE       AttributeError: 'NoneType' object has no attribute 'cursor'\n\ntests\\Part_2\\test_Part_2.py:39: AttributeError"}, "teardown": {"duration": 0.00025679999999994596, "outcome": "passed"}}, {"nodeid": "tests/Part_2/test_Part_2.py::test_Part_2_data", "lineno": 45, "outcome": "failed", "keywords": ["ds-sa-db-api", "tests/Part_2/test_Part_2.py", "test_Part_2_data"], "setup": {"duration": 0.0003066999999999931, "outcome": "passed"}, "call": {"duration": 0.00027169999999998584, "outcome": "failed", "crash": {"path": "C:\\Users\\Coke\\playground\\Section3\\ds-sa-db-api\\tests\\Part_2\\test_Part_2.py", "lineno": 48, "message": "AttributeError: 'NoneType' object has no attribute 'cursor'"}, "traceback": [{"path": "tests\\Part_2\\test_Part_2.py", "lineno": 48, "message": "AttributeError"}], "longrepr": "def test_Part_2_data():\n    \n>       cursor = conn.cursor()\nE       AttributeError: 'NoneType' object has no attribute 'cursor'\n\ntests\\Part_2\\test_Part_2.py:48: AttributeError"}, "teardown": {"duration": 0.0006041999999999437, "outcome": "passed", "stdout": "\n        -------------------------------------\n        Tearing down...\n        -------------------------------------\n        \n"}}, {"nodeid": "tests/Part_3/test_Part_3.py::test_Part_3_exist", "lineno": 13, "outcome": "failed", "keywords": ["test_Part_3_exist", "ds-sa-db-api", "tests/Part_3/test_Part_3.py"], "setup": {"duration": 0.0005222000000000282, "outcome": "passed", "stdout": "\n        -------------------------------------\n        Setting up...\n        -------------------------------------\n        \n"}, "call": {"duration": 0.0002763000000000071, "outcome": "failed", "crash": {"path": "C:\\Users\\Coke\\playground\\Section3\\ds-sa-db-api\\tests\\Part_3\\test_Part_3.py", "lineno": 15, "message": "assert False"}, "traceback": [{"path": "tests\\Part_3\\test_Part_3.py", "lineno": 15, "message": "AssertionError"}], "longrepr": "def test_Part_3_exist():\n>       assert table_exist\nE       assert False\n\ntests\\Part_3\\test_Part_3.py:15: AssertionError"}, "teardown": {"duration": 0.0002497999999999667, "outcome": "passed"}}, {"nodeid": "tests/Part_3/test_Part_3.py::test_Part_3_schema", "lineno": 21, "outcome": "failed", "keywords": ["ds-sa-db-api", "test_Part_3_schema", "tests/Part_3/test_Part_3.py"], "setup": {"duration": 0.000298400000000032, "outcome": "passed"}, "call": {"duration": 0.0002359000000000666, "outcome": "failed", "crash": {"path": "C:\\Users\\Coke\\playground\\Section3\\ds-sa-db-api\\tests\\Part_3\\test_Part_3.py", "lineno": 29, "message": "AttributeError: 'NoneType' object has no attribute 'cursor'"}, "traceback": [{"path": "tests\\Part_3\\test_Part_3.py", "lineno": 29, "message": "AttributeError"}], "longrepr": "def test_Part_3_schema():\n        fields = {\n            ('AlbumId','INTEGER', 1, 1),\n            ('Title','NVARCHAR(160)', 0, 0),\n            ('ArtistId','INTEGER', 0, 0)\n        }\n    \n>       cursor = conn.cursor()\nE       AttributeError: 'NoneType' object has no attribute 'cursor'\n\ntests\\Part_3\\test_Part_3.py:29: AttributeError"}, "teardown": {"duration": 0.00024449999999998084, "outcome": "passed"}}, {"nodeid": "tests/Part_3/test_Part_3.py::test_Part_3_count", "lineno": 36, "outcome": "failed", "keywords": ["ds-sa-db-api", "test_Part_3_count", "tests/Part_3/test_Part_3.py"], "setup": {"duration": 0.00022860000000002323, "outcome": "passed"}, "call": {"duration": 0.0002162999999999471, "outcome": "failed", "crash": {"path": "C:\\Users\\Coke\\playground\\Section3\\ds-sa-db-api\\tests\\Part_3\\test_Part_3.py", "lineno": 39, "message": "AttributeError: 'NoneType' object has no attribute 'cursor'"}, "traceback": [{"path": "tests\\Part_3\\test_Part_3.py", "lineno": 39, "message": "AttributeError"}], "longrepr": "def test_Part_3_count():\n    \n>       cursor = conn.cursor()\nE       AttributeError: 'NoneType' object has no attribute 'cursor'\n\ntests\\Part_3\\test_Part_3.py:39: AttributeError"}, "teardown": {"duration": 0.00024209999999991183, "outcome": "passed"}}, {"nodeid": "tests/Part_3/test_Part_3.py::test_Part_3_data", "lineno": 45, "outcome": "failed", "keywords": ["ds-sa-db-api", "test_Part_3_data", "tests/Part_3/test_Part_3.py"], "setup": {"duration": 0.0002299000000000051, "outcome": "passed"}, "call": {"duration": 0.0003370999999999791, "outcome": "failed", "crash": {"path": "C:\\Users\\Coke\\playground\\Section3\\ds-sa-db-api\\tests\\Part_3\\test_Part_3.py", "lineno": 48, "message": "AttributeError: 'NoneType' object has no attribute 'cursor'"}, "traceback": [{"path": "tests\\Part_3\\test_Part_3.py", "lineno": 48, "message": "AttributeError"}], "longrepr": "def test_Part_3_data():\n    \n>       cursor = conn.cursor()\nE       AttributeError: 'NoneType' object has no attribute 'cursor'\n\ntests\\Part_3\\test_Part_3.py:48: AttributeError"}, "teardown": {"duration": 0.0005321000000000353, "outcome": "passed", "stdout": "\n        -------------------------------------\n        Tearing down...\n        -------------------------------------\n        \n"}}, {"nodeid": "tests/Part_4/test_Part_4.py::test_passenger_table", "lineno": 4, "outcome": "error", "keywords": ["ds-sa-db-api", "tests/Part_4/__init__.py", "test_passenger_table", "test_Part_4.py"], "setup": {"duration": 0.001225900000000002, "outcome": "failed", "crash": {"path": "C:\\Users\\Coke\\playground\\Section3\\ds-sa-db-api\\tests\\Part_4\\conftest.py", "lineno": 17, "message": "AttributeError: module 'src.Part_4' has no attribute 'connection'"}, "traceback": [{"path": "tests\\Part_4\\conftest.py", "lineno": 17, "message": "AttributeError"}], "stdout": "\n        -------------------------------------\n        Setting up...\n        -------------------------------------\n        \n", "longrepr": "@pytest.fixture(scope=\"module\")\n    def db_connection():\n        conn = ''\n>       if type(Part_4.connection) != str:\nE       AttributeError: module 'src.Part_4' has no attribute 'connection'\n\ntests\\Part_4\\conftest.py:17: AttributeError"}, "teardown": {"duration": 0.0002763000000000071, "outcome": "passed"}}, {"nodeid": "tests/Part_4/test_Part_4.py::test_titanic_data", "lineno": 30, "outcome": "error", "keywords": ["ds-sa-db-api", "test_titanic_data", "tests/Part_4/__init__.py", "test_Part_4.py"], "setup": {"duration": 0.00033050000000001134, "outcome": "failed", "crash": {"path": "C:\\Users\\Coke\\playground\\Section3\\ds-sa-db-api\\tests\\Part_4\\conftest.py", "lineno": 17, "message": "AttributeError: module 'src.Part_4' has no attribute 'connection'"}, "traceback": [{"path": "tests\\Part_4\\conftest.py", "lineno": 17, "message": "AttributeError"}], "longrepr": "@pytest.fixture(scope=\"module\")\n    def db_connection():\n        conn = ''\n>       if type(Part_4.connection) != str:\nE       AttributeError: module 'src.Part_4' has no attribute 'connection'\n\ntests\\Part_4\\conftest.py:17: AttributeError"}, "teardown": {"duration": 0.000545600000000035, "outcome": "passed", "stdout": "\n        -------------------------------------\n        Tearing down...\n        -------------------------------------\n        \n"}}]}